src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -241,5 +280,7 @@ private ConnectionPool(ConnectionSettings cs)
 		readonly SemaphoreSlim m_sessionSemaphore;
 		readonly LinkedList<MySqlSession> m_sessions;
 		readonly ConnectionSettings m_connectionSettings;
+		readonly Dictionary<int, WeakReference<MySqlSession>> m_leasedSessions;
+		int m_lastId;
```

Comment:
As long as it wraps around I think it's fine as `int`. I wasn't sure​ if C# throws an OverflowException.